universal-login-example/src/services/Threebox.js

```diff
@@ -0,0 +1,22 @@
+import Box from '3box';
+
+class ThreeBox {
+  constructor(provider) {
+    console.log(Box);
+    this.provider = provider;
+    this.enabled = false;
+    this.box = null;
+    this.addresses = [];
+  }
+
+  async enable() {
+    this.addresses = await window.ethereum.enable();
```

Comment:
This is where you probably need to change. There is no concept of window.ethereum here–but then maybe it should have.

@marekkirejczyk what if instead of having to change 3box here, don't we change the SDK to provide an ethereum object by default and it will pipe through all the transaction requests to either a meta transaction for the identity or a simple signature by one of the local keys.

Reply:
@alexvandesande, we should integrate this instead: https://github.com/MetaMask/mascara

Reply:
I'm sorry, Chris, there seems to be a misunderstanding on what Universal Logins is supposed to be. We are not integrating Mascara. Universal logins sorts of makes Mascara (or Metamask, Mist, etc) redundant (or only as an optional add-on for extra security).

The app already has a local private key. We should use that for signing messages instead.

Reply:
This is my bad.
I got confused from the 3box-js documentation. I assumed that metamask was necessary. I was aware of what the repository was trying to do... I was confused on how to make the implementation without Metamask.

**From the 3box-js documentation:**

> ### Get, set, and remove data
> To get or modify data in a user's 3Box, first open their 3Box by calling the openBox method. This method prompts the user to authenticate your dapp and returns a promise with a threeBox instance. You can only set, get, and remove data of users that are currently interacting with your dapp. Below ethereumProvider refers to the object that you would get from web3.currentProvider, or window.ethereum.

When you prompt the `.openBox()` method. It opens a dialog in Metamask to allow the dApp to work. In this case, we just need to pop in the provider for the existing web3 instance.

Reply:
Chris, you are correct, that's how you use 3Box with a ethereum enabled browser. Your post convinced me that, instead of changing the 3box code, we should instead thrive to have a `web3.currentProvider` or `window.ethereum` objects on our app that simulates what the tools expect from metamask

Reply:
@alexvandesande, I'm working on something now that ideally will be able to work Metamask or Mascara. I will give you an update if there are problems. Or a PR for review if no issues.